x-pack/plugins/security_solution/server/endpoint/services/actions/clients/endpoint/endpoint_actions_client.ts

```diff
@@ -122,29 +130,40 @@ export class EndpointActionsClient extends ResponseActionsClientImpl {
       }
     }
 
+    // Append warning message to comment if there are invalid agents
+    const commentMessage = actionReq.comment ? actionReq.comment : '';
+    const warningMessage = `(WARNING: ${getInvalidAgentsWarning(validatedAgents.invalid)})`;
+    const comment = validatedAgents.invalid.length
+      ? commentMessage
+        ? `${commentMessage}. ${warningMessage}`
+        : warningMessage
+      : actionReq.comment;
```

Comment:
(Optional) Could be simplified (and avoid potentially calling `getInvalidAgentsWarning()` unnecessary) : 

```typescript
const comment = actionReq.comment ?? '' + 
  validatedAgents.invalid.length
  ? `${actionReq.comment ? '. ' : ''}(WARNING: ${getInvalidAgentsWarning(validatedAgents.invalid)})`
  : ''

```

Reply:
I'm reusing `getInvalidAgentsWarning` in the `checkAgents` as well. So I'll leave it as it is for now.